src/telemetry/telemetry.c

```diff
@@ -505,6 +505,9 @@ ts_telemetry_main(const char *host, const char *path, const char *service)
 	bool started = false;
 	const char *volatile json = NULL;
 
+	if (!ts_telemetry_on())
```

Comment:
Should we even get here if telemetry is off? Maybe this should be an assertion, since it seems to be a bug to start telemetry if it is off.

Reply:
i think it's important to have this here as this is the code that is concerned with telemetry. we could also assert this but the assertion would fail in our test code as there are ways to circumvent the scheduler and some of our tests do